var/spack/repos/builtin/packages/py-wradlib/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyWradlib(PythonPackage):
+    """wradlib is designed to assist you in the most important steps of
+    processing weather radar data. These may include: reading common data
+    formats, georeferencing, converting reflectivity to rainfall intensity,
+    identifying and correcting typical error sources (such as clutter or
+    attenuation) and visualising the data."""
+
+    homepage = "https://docs.wradlib.org"
+    url      = "https://pypi.io/packages/source/w/wradlib/wradlib-1.5.0.tar.gz"
+
+    version('1.5.0', sha256='9bf0742d7235ea830e83c2269f6b5d1afd83d92696efce0a7bcdb0c4f6604784')
+
+    depends_on('py-setuptools', type='build')
+
+    # https://docs.wradlib.org/en/stable/index.html
+    conflicts('^python@:2.999', when='@1.3:')
+
+    # recommended versions from https://docs.wradlib.org/en/stable/installation.html#dependencies
+    depends_on('py-numpy@1.16:', type=('build', 'run'))
+    depends_on('py-matplotlib@3.0.2:', type=('build', 'run'))
+    depends_on('py-scipy@1.2.0:', type=('build', 'run'))
+    depends_on('py-h5py@2.9:', type=('build', 'run'))
+    depends_on('py-netcdf4@1.4.2:', type=('build', 'run'))
+    depends_on('py-xarray@0.11.3:', type=('build', 'run'))
+    depends_on('py-xmltodict@0.11:', type=('build', 'run'))
+    depends_on('py-pygdal@2.4.0:', type=('build', 'run'))
+    depends_on('py-semver', type=('build', 'run'))
+    depends_on('py-deprecation', type=('build', 'run'))
+    depends_on('py-requests', type=('build', 'run'))
```

Comment:
Are you sure this is required?

Reply:
- semver is used by setup.py
- requests is used by dem.py

I'm not sure about py-deprecation. It's listed in the requirements.txt but after looking into the source I did not find any import of this package. I'll try to build and run wradlib without it.

Reply:
I usually only trust what is explicitly listed in `install_requires` in the `setup.py`. If it's listed, setuptools will try to download and install it regardless of if it's actually used. If it's not listed, that's a bug in the `setup.py` that should be reported to the developers.

Reply:
`setup.py` sets `install_requires` up by simply reading `requirements.txt`, therefore currently py-deprecation is still necessary. I'll report this issue to the developers so that this dependency can be dropped in future versions.

Reply:
For information only: I just checked the development version on github...it differs from the release on pypi. Amongst others the github version still uses the `deprecation` package. So currently it's okay to omit this dependency.